File: src/main/java/ru/herobrine1st/fusion/api/module/IModule.java
package ru.herobrine1st.fusion.api.module;

import net.dv8tion.jda.api.JDA;
import ru.herobrine1st.fusion.api.manager.CommandManager;

public interface IModule {
    void registerCommands(CommandManager commandManager);
    void registerListener(JDA jda);
}

File: src/main/kotlin/ru/herobrine1st/fusion/internal/module/IModule.kt
package ru.herobrine1st.fusion.internal.module

import net.dv8tion.jda.api.JDA
import net.dv8tion.jda.api.requests.restaction.CommandListUpdateAction

interface IModule {
    fun registerCommands(commandListUpdateAction: CommandListUpdateAction)
    fun registerListener(jda: JDA)
}